server/applicationPattern/applicationPattern/onfModel/services/ForwardingConstructConfigurationServices.test.js

```diff
@@ -0,0 +1,376 @@
+    jest.spyOn(HttpClientInterface, 'getApplicationNameAsync').mockReturnValueOnce('RegistryOffice');
+    jest.spyOn(HttpClientInterface, 'getReleaseNumberAsync').mockReturnValueOnce('2.0.1');
+    jest.spyOn(LogicalTerminationPoint, 'getServerLtpListAsync').mockReturnValue(["ro-2-0-1-op-c-im-ol-1-0-0-003"]);
+    jest.spyOn(HttpClientInterface, 'getApplicationNameAsync').mockReturnValue('RegistryOffice')
+    jest.spyOn(HttpClientInterface, 'getReleaseNumberAsync').mockReturnValue('2.0.2');
+    jest.spyOn(fcPort, 'setLogicalTerminationPointAsync').mockImplementation(() => true);
+    jest.spyOn(ForwardingConstruct, 'addFcPortAsync').mockImplementation(() => true);
+
+    const res = await ForwardingConstructConfigurationServices.configureForwardingConstructAsync(
+      operationServerName, forwardingConfigurationInputList);
+    
+    const forwardingConstructConfigurationStatusList = [new ConfigurationStatus("ro-2-0-1-op-fc-is-007", '', true)];
+    const fcPortConfigurationStatusList = [
+      new ConfigurationStatus("ro-2-0-1-op-fc-is-007", '204', true),
+      new ConfigurationStatus("ro-2-0-1-op-fc-is-007", '204', true)
+    ];
+    expect(res).toStrictEqual(new ForwardingConstructConfigurationStatus(
+      forwardingConstructConfigurationStatusList,
+      fcPortConfigurationStatusList
+      ));
+    
+  });
+
+  test("configureForwardingConstructAsync - successful", async () => {
+    const operationServerName = '/v1/embed-yourself';
+
+    jest.spyOn(OperationServerInterface, 'getOperationServerUuidAsync').mockImplementation(() => 'ro-2-0-1-op-s-bm-001');
+    jest.spyOn(ForwardingDomain, 'getForwardingConstructForTheForwardingNameAsync').mockImplementation(() => fc);
+    jest.spyOn(fcPort, 'setLogicalTerminationPointAsync').mockImplementation(() => true);
+
+    const res = await ForwardingConstructConfigurationServices.configureForwardingConstructAsync(
+      operationServerName, forwardingConfigurationInputList);
+    
+    const forwardingConstructConfigurationStatusList = [new ConfigurationStatus("ro-2-0-1-op-fc-bm-106", '', false)];
+    const fcPortConfigurationStatusList = [
+      new ConfigurationStatus("ro-2-0-1-op-fc-bm-106", '200', true),
+      new ConfigurationStatus("ro-2-0-1-op-fc-bm-106", '200', true)
+    ];
+    expect(res).toStrictEqual(new ForwardingConstructConfigurationStatus(forwardingConstructConfigurationStatusList, fcPortConfigurationStatusList));
+    
+  });
+
+  test("configureForwardingConstructAsync - failed to get OperationServerUuidAsync", async () => {
+    const operationServerName = '/v1/embed-yourself';
+    const mockError = { message: 'Something bad happened' };
+
+    jest.spyOn(OperationServerInterface, 'getOperationServerUuidAsync').mockImplementation(() => {
+      return Promise.reject(mockError);
+    });
+
+    const res = ForwardingConstructConfigurationServices.configureForwardingConstructAsync(
+      operationServerName, forwardingConfigurationInputList);
+    return res.then()
```

Comment:
the '.then()' part is useless. Also although it works, this is not the way it should be tested using jest. Take a look at: https://jestjs.io/docs/expect#rejects 
`test('rejects to octopus', async () => {
  await expect(Promise.reject(new Error('octopus'))).rejects.toThrow('octopus');
});`